l10n_it_fatturapa_in/models/attachment.py

```diff
@@ -80,45 +91,109 @@ def get_xml_string(self):
 
     @api.multi
     def recompute_xml_fields(self):
-        self._compute_xml_data()
+        # Pretend the attachment has been modified
+        # and trigger a recomputation:
+        # this recomputes all fields whose value
+        # is extracted from the attachment
+        self.modified(['ir_attachment_id'])
+        self.recompute()
+
         self._compute_registered()
 
+    @api.multi
+    def get_invoice_obj(self):
+        self.ensure_one()
+        invoice_obj = False
+        try:
+            xml_string = self.get_xml_string()
+            invoice_obj = fatturapa.CreateFromDocument(xml_string)
+        except Exception as e:
+            error_msg = \
+                _("Impossible to parse XML for {att_name}: {error_msg}") \
+                .format(
+                    att_name=self.display_name,
+                    error_msg=e,
+                )
+            _logger.error(error_msg)
+            self.e_invoice_parsing_error = error_msg
+        else:
+            self.e_invoice_parsing_error = False
+        return invoice_obj
+
     @api.multi
     @api.depends('ir_attachment_id.datas')
-    def _compute_xml_data(self):
+    def _compute_is_self_invoice(self):
         for att in self:
-            if not att.registered:
-                wiz_obj = self.env['wizard.import.fatturapa'] \
-                    .with_context(from_attachment=att)
-                fatt = wiz_obj.get_invoice_obj(att)
-                cedentePrestatore = fatt.FatturaElettronicaHeader.CedentePrestatore
-                partner_id = wiz_obj.getCedPrest(cedentePrestatore)
-                att.xml_supplier_id = partner_id
-                att.invoices_number = len(fatt.FatturaElettronicaBody)
-                att.invoices_total = 0
-                att.is_self_invoice = False
-                invoices_date = []
+            fatt = att.get_invoice_obj()
+            att.is_self_invoice = False
+            if fatt:
```

Comment:
Farei `if not attachment.e_invoice_parsing_error`

Reply:
Vedi https://github.com/OCA/l10n-italy/pull/2788#discussion_r904715352